webapp-django/crashstats/api/views.py

```diff
@@ -133,23 +111,12 @@ def clean(self):
 )
 
 
-def has_permissions(user, permissions):
-    for permission in permissions:
-        if not user.has_perm(permission):
-            return False
-    return True
-
-
 def is_valid_model_class(model):
-    try:
-        return (
-            issubclass(model, models.SocorroMiddleware) and
-            model is not models.SocorroMiddleware and
-            model is not supersearch_models.ESSocorroMiddleware
-        )
-    except TypeError:
-        # issubclass throws a TypeError if the model is not a class.
-        return False
+    return (
+        issubclass(model, models.SocorroMiddleware) and
```

Comment:
I think you're right on. +1

I think this is leftover from when I rewrote the discovery code for API models so a string couldn't inadvertently get passed in here which would kick up the `TypeError`. I think it happened when someone was fuzzing the `/api` endpoint and passing in things that weren't model names. Or something like that.

Reply:
> I think this is leftover from when I rewrote the discovery code for API models so a string couldn't inadvertently get passed in here which would kick up the TypeError. I think it happened when someone was fuzzing the /api endpoint and passing in things that weren't model names. Or something like that.

I figured this our shortly after merging. The code looks for an attribute in a list of modules and passes that to ``is_valid_model_class``. For example, ``models.py`` has an ``import time``, so ``/api/time`` will pass the standard library module ``time`` to ``is_valid_model_class``, which will fail with a ``TypeError``. Oh well, here comes the next PR...

Reply:
I thought I rewrote it because I remember feeling unenthused about string -> model lookups. Maybe I rewrote it but never landed it for some reason. Oh well. :(